modules/afprog/afprog.c

```diff
@@ -145,13 +217,25 @@ afprogram_popen(AFProgramProcessInfo *process_info, GIOCondition cond, gint *fd)
       close(msg_pipe[0]);
       close(msg_pipe[1]);
 
+      _closing_all_fd_expect(sync_pipe, 2);
```

Comment:
if you dup2() the writer side of sync_pipe to fd 3, you can simply close all fds above 3, no need for the "ignore" support.

Reply:
and if you iterate backwards (e.g. from large fds to small ones), that could even do the closing of the sync pipe.